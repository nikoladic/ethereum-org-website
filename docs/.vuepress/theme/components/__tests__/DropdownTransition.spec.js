import { shallowMount } from '@vue/test-utils'
import DropdownTransition from "@/theme/components/DropdownTransition"

describe('DropdownTransition', () => {
  test('render slots correctly', () => {
    const wrapper = shallowMount(DropdownTransition, {
      slots: {
        default: '<div class="box"><p>Ethereum</p></div>'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })


  test('render transition after enter', done => {
    const wrapper = shallowMount(DropdownTransition, {
      slots: {
        default: '<div />'
      }
    })
    wrapper.trigger("transitionstart")
    wrapper.vm.$nextTick(() => {
      expect(wrapper.element).toMatchSnapshot()
      done()
    })
  })
})
