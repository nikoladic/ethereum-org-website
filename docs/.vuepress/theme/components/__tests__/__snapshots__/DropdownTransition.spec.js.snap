// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownTransition render slots correctly 1`] = `
<div
  class="box"
  name="dropdown"
>
  <p>
    Ethereum
  </p>
</div>
`;

exports[`DropdownTransition render transition after enter 1`] = `
<div
  name="dropdown"
/>
`;
